checker.c

```diff
@@ -1,21 +1,145 @@
+ =======               DEFINES & MACROS FOR GENERAL PURPOSE              =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                       CONSTANTS  &  TYPES                       =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                VARIABLES & MESSAGES & RESSOURCEN                =======
+ ==============================================================================*/
+static BMS_parameters_s BMS_parameters_attributes = {
+		false,
+		false,
+		false,
+		false
+};
+/*=============================================================================
+ =======                              METHODS                            =======
+ =============================================================================*/
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_checkAttribute_OutOfRange
+ */
+/*!    \brief   Function to check the range of parameters
+ *
+ *
+ *     \param    	bms_attribute -> the parameter for which range needs to be checked
+					attribute_Min_Value -> Lower range of the parameter
+					attribute_Max_Value -> Upper range of the parameter
+					param[]	-> Parameter name to be printed
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+bool BMS_checkAttribute_OutOfRange(float bms_attribute, float attribute_Min_Value, float attribute_Max_Value, char param[])
+{
+	bool retAtributeStatus = false;
+	if(bms_attribute < attribute_Min_Value || bms_attribute > attribute_Max_Value) 
+	{
+		BMS_printParameterStatus_As_OutOfRange(param);
+		retAtributeStatus = true;
+		BMS_setBMSStatus(true);									
+	}	
+	
+	return retAtributeStatus;
+}
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_printParameterStatus_As_OutOfRange
+ */
+/*!    \brief   Function to print the parameter out of range
+ *
+ *
+ *     \param    	param[] -> the parameter will be printed
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+void BMS_printParameterStatus_As_OutOfRange(char param[])
+{
+	printf("%s out of range!\n",param);
+}
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_setBMSStatus
+ */
+/*!    \brief   Function to set BMS status 
+ *
+ *
+ *     \param    	bms_Status -> will be set as TRUE if any parameter is out of range
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+void BMS_setBMSStatus(bool bms_Status)
+{
+	BMS_parameters_attributes.bms_Status_b = bms_Status;			/*If any parameter out of range, set bms_Status_b as TRUE*/
```

Comment:
Instead of writing the comment, try a better name for `bms_Status_b`. Like `battery_needs_attention`. Otherwise we need to refer back to this location every time we see the variable.